packages/cyberstorm/src/components/DropDown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, ReactNode, useState } from "react";
+import styles from "./componentStyles/DropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { DropDownItemProps } from "./DropDownItem";
+
+type DropDownProps = {
+  icon?: ReactNode;
+  defaultOpen?: boolean;
+  options?: DropDownOptions;
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultSelectedOptionKey?: number;
+};
+
+export type DropDownOption = {
+  label: string;
+  content?: ReactElement<DropDownItemProps>;
+};
+export type DropDownOptions = Map<number | string, DropDownOption>;
```

Comment:
I think I prefer the earlier approach where `DropDownOption` included a `value`. Now the value is "hidden" as the key in the Map, making it sort of implicit rather than explicit. I guess this could be improved by typing this as `Map<DropDownOptionValue, DropDownOption>`, but that still feels awkward in unintuitive to *me* as a developer. I think it would also mean that we need to do (more) processing to get the values returned from the backend into a format that this component accepts. @MythicManiac could chip in with a second opinion since I'm not sure if it's just my "not invented by me" bias speaking.

To clarify, I suggest the beginning of the component should be something like this:
```typescript
export type DropDownOption = {
  value: string | number;
  content: typeof DropDownItem;
};

type DropDownProps = {
  colorScheme?: "default" | "defaultDark" | "primary";
  defaultOpen?: boolean;
  defaultValue?: string | number;
  icon?: ReactNode;
  options?: DropDownOption[];
};

export const DropDown: React.FC<DropDownProps> = (props) => {
  const { colorScheme, defaultOpen, defaultValue, icon, options } = props;

  const initialValue = options?.find((o) => o.value === defaultValue) ?? null;
  // ...rest of the component here...
```

Note that `DropDownOption.label` was yeeted since it wasn't used anymore.